daemon/stats.go

```diff
@@ -12,31 +13,39 @@ func (daemon *Daemon) ContainerStats(name string, stream bool, out io.Writer) er
 	if err != nil {
 		return err
 	}
-	var pre_cpu_stats types.CpuStats
-	for first_v := range updates {
-		first_update := first_v.(*execdriver.ResourceStats)
-		first_stats := convertToAPITypes(first_update.Stats)
-		pre_cpu_stats = first_stats.CpuStats
-		pre_cpu_stats.SystemUsage = first_update.SystemUsage
-		break
-	}
-	enc := json.NewEncoder(out)
-	for v := range updates {
+
+	var preCpuStats types.CpuStats
+	getStat := func(v interface{}) *types.Stats {
 		update := v.(*execdriver.ResourceStats)
-		ss := convertToAPITypes(update.Stats)
-		ss.PreCpuStats = pre_cpu_stats
+		ss := convertStatsToAPITypes(update.Stats)
+		ss.PreCpuStats = preCpuStats
 		ss.MemoryStats.Limit = uint64(update.MemoryLimit)
 		ss.Read = update.Read
 		ss.CpuStats.SystemUsage = update.SystemUsage
-		pre_cpu_stats = ss.CpuStats
-		if err := enc.Encode(ss); err != nil {
+		preCpuStats = ss.CpuStats
+		return ss
+	}
+
+	enc := json.NewEncoder(out)
+
+	if !stream {
```

Comment:
twice because it needs to compare CPU stat values to provide meaningful information.

preCpuStats is actually set in `getStat()` above.
I think I can actually remove lines 33 and 34 here since it's setting that above.  Not sure why that was being done.